### ISSUE TITLE
[Bug]: "smart" HTTP error when trying to pull, push, or clone 

### ISSUE THREAD

**Issue by vadtec** [denolehov/obsidian-git#594]
### Describe the bug

When trying to pull, push or clone from a remote server (used successfully for months with obsidian git) I am now getting a "smart" HTTP error. 

### Relevant errors (if available) from notifications or console (`CTRL+SHIFT+I`)

```shell
This is on Android. I'll attach screenshots.
```

### Steps to reproduce

Pull or push existing repo that has successfully done so in the past. Or, clone any repo from the same source. 

### Expected Behavior

Operations should pull, push, or clone. 

### Addition context

![Screenshot_20230907_070841_Obsidian](https://github.com/denolehov/obsidian-git/assets/228679/0b444cca-b771-4357-82e9-01e83ce7dea7)
![Screenshot_20230907_095752_Obsidian](https://github.com/denolehov/obsidian-git/assets/228679/91430758-f32f-40d1-b135-e31dafbf29f9)

For reference, this server has roughly 30 end users committing to numerous repos anything from every 5 minutes to every few hours using Linux, Mac, Windows, termux on Android, etc. I am only having this issue with obsidian git as outlined above. 

### Operating system

Android

### Installation Method

None

### Plugin version

2.22.0

**Comment by vadtec** [comment]
![Screenshot_20230907_103346_Obsidian](https://github.com/denolehov/obsidian-git/assets/228679/6efddcd2-a3f7-485c-bdc3-d4c7722feee9)

**Comment by vadtec** [comment]
So, I downloaded the latest release of main.js and found the offending line of code on line 20988. Not really sure what the issue is, or where it exists in the unbundled code, but I at least know it does exist.

**Comment by vadtec** [comment]
Has anyone looked into this issue? I don't mind having to use termux, but I'd much rather use obsidian without having to jump through hoops.

**Comment by vadtec** [comment]
So is this issue being looked into? Reverting really isn't an option. Using termux is a headache. I don't expect perfection, but a little communication would be appreciated.

What exploration of the code I've done hasn't yielded anything useful other than I know where the issue ultimately is thrown. Not being familiar with the code or build process has me at a major disadvantage.

So does anyone have any info on this?

**Comment by vadtec** [comment]
I moved the repo so I could manually install older versions of this plugin. So far, I've gone all the way back to 2.18.whatever and I'm still getting the same error. When I installed this back in late May, it was working. Nothing has changed on the server in question, so some underlying issue is going on. Just passing it along.

**Comment by vadtec** [comment]
And just like that.... It starts working. Restarted the webserver.... Yep. Restarted the entire server... Yep. Upgraded all relevant software and libraries... Yep. Hell... Restarted two firewalls... Yep. Upgraded Android device... Yep. This is the shit that drives me crazy as an IT guy.

Still not thrilled with the lack of communication...

**Comment by Vinzent03** [comment]
Sorry for not responding, but haven't seen that error before and haven't the time to really look into that. I'm still doing that in my free time. Great, it's solved now.

**Comment by vadtec** [comment]
@nikelborm 

How quaint...a keyboard warrior. The assumptions you make are amusing. Like most other people in this world, you've made baseless assumptions about what another person may or may not have done outside of your knowledge. And a full 110 days after the issue was closed no less. Way to get that social justice!

Worry not, ye of less acumen, I do not hold you accountable. Alas, such lack of foresight is so common place in this era it wouldn't surprise me if historians of the future mark these recent years as a downturn in human intellect.

Asinine comments aside, perhaps next time resisting the urge to be a keyboard warrior is the better part of discretion. You took my dissatisfaction with a lack of communication, gods forbid even in an open source project, and decided you knew me, apparently better than I know myself. Explain to others how what I said justifies anything you said. The only true statement you said is this:

> This is open source and you can't demand anything from it's authors.

Strange, I don't see any demands being made. What I do see is an end user providing every last shred of information they can, including digging into a completely unknown code base to help identify the issue if possible, in an attempt to help solve an issue that suddenly appeared. Somehow, you have conflated reporting an issue and expressing dissatisfaction as being demanding and unhelpful. This entire thread is the antithesis of your misguided assumptions.

May your programming career be filled with off-by-one errors and your coffee taste of staleness.

P.S. My response to you is for posterity. I don't really give a damn what you or anyone else thinks about me. I'll get to live in your head, rent free, at least for a little bit. And while I'm there, I'm going to enjoy myself.

Here's some unsolicited advice: you are not so important in this world that others are compelled to listen when you speak.

Thanks for the chuckles, the office guys are loving it.

**Comment by jeffeb3** [comment]
I am getting a SmartHttpError myself. Very similar circumstances. Private server (running GitBucket at home), Termux git commands work fine, Obsidian-git from Linux works fine. SmartHttpError expected something similar, but in my case, it got and html page. I can only read the top of it.

Are there logs somewhere with the rest of the error message?

Does Obsidian-Git come with its own git binary? Which binary does it use? The termux one works (but it could be something different with the environment).

My credentials are saved in Termux with the cached credentials command (they are in ~/.git-credentials). I tried adding the username/password to Obsidian-Git.

I have rebooted GitBucket. No change on my end. Luckily for me I am the only one that uses it.

Thanks in advance.
